base-plugin/src/main/groovy/com/github/jrubygradle/internal/JRubyExecUtils.groovy

```diff
@@ -176,4 +207,77 @@ class JRubyExecUtils {
         }
     }
 
+    /**
+     * Prepare the Ruby and Java dependencies for the configured configuration
+     *
+     * This method will determine the appropriate dependency overwrite behavior
+     * from the Gradle invocation. In effect, if the --refresh-dependencies flag
+     * is used, already installed gems will be overwritten.
+     *
+     * @param project The associated Gradle project.
+     * @param gemWorkDir THe GEM unpack/working directory.
+     * @param jruby The associated JRuby project or task extension.
+     * @param gemConfiguration Configuration which contains GEMs for unpacking.
+     * @param overwrite Overwrite mode.
+     *
+     * @since 2.0
+     */
+    static void prepareDependencies(
```

Comment:
This code was moved in here from the JRubyExec task. It makes it easier to manage. re-use and test the GEM prep code from outside a task.